### ISSUE TITLE
Cannot call this method while RecyclerView is computing a layout or scrolling

### ISSUE THREAD

**Issue by paour** [Gericop/Android-Support-Preference-V7-Fix#84]
This is similar to #44, but I'm still getting it with the latest versions of the libraries.

```
Fatal Exception: java.lang.IllegalStateException: Cannot call this method while RecyclerView is computing a layout or scrolling
       at android.support.v7.widget.RecyclerView.assertNotInLayoutOrScroll(RecyclerView.java:2586)
       at android.support.v7.widget.RecyclerView$c.onItemRangeChanged(RecyclerView.java:4951)
       at android.support.v7.widget.RecyclerView$a.a(RecyclerView.java:11371)
       at android.support.v7.widget.RecyclerView$Adapter.notifyItemChanged(RecyclerView.java:6678)
       at android.support.v7.preference.f.a(PreferenceGroupAdapter.java:264)
       at android.support.v7.preference.Preference.a_(Preference.java:1070)
       at android.support.v7.preference.TwoStatePreference.e(TwoStatePreference.java:85)
       at android.support.v7.preference.SwitchPreferenceCompat$a.onCheckedChanged(SwitchPreferenceCompat.java:60)
       at android.widget.CompoundButton.setChecked(CompoundButton.java:166)
       at android.support.v7.widget.SwitchCompat.setChecked(SwitchCompat.java:1060)
       at com.takisoft.fix.support.v7.preference.SwitchPreferenceCompat.a(SwitchPreferenceCompat.java:44)
       at android.support.v7.preference.f.a(PreferenceGroupAdapter.java:361)
       at android.support.v7.preference.f.onBindViewHolder(PreferenceGroupAdapter.java:42)
       at android.support.v7.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:6354)
       at android.support.v7.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:6387)
       at android.support.v7.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:5343)
       at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5606)
       at android.support.v7.widget.s.a(GapWorker.java:285)
       at android.support.v7.widget.s.a(GapWorker.java:342)
       at android.support.v7.widget.s.b(GapWorker.java:358)
       at android.support.v7.widget.s.a(GapWorker.java:365)
       at android.support.v7.widget.s.run(GapWorker.java:396)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6776)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
```

**Comment by Gericop** [comment]
Can you reproduce this issue multiple times? Because it seems to be something that happens when scrolling the list but I just couldn't reproduce with the sample app no matter what. If you had a sample project for this, I'd appreciate it.

**Comment by paour** [comment]
Unfortunately I haven't been able to reproduce it myself, but on Crashlytics I'm getting multiple reports per day from different users with this exception.

There isn't a clear pattern in terms of Android version numbers or devices and I haven't been able to narrow it down to a specific preference. Because this is happening in code I don't control, I haven't been able to instrument it.

**Comment by Gericop** [comment]
I think I'll just remove the custom `SwitchPreferenceCompat` class entirely as it seems to me that the no-animation bug is fixed now.

**Comment by Gericop** [comment]
Resolved in v**26.0.0.1**
